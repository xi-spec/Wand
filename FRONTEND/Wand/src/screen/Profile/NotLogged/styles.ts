import { StyleSheet } from 'react-native';
import { COLORS, FONTS, SIZE } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  line: {
    width: '80%',
    borderBottomWidth: 1,
    borderBottomColor: COLORS.lightGrey,
    marginLeft: 5,
    marginRight: 5

  },
  text: {
    fontSize: SIZE.primary,
    fontFamily: FONTS.logoRegular,
    marginVertical: 20
  },
  buttonContainer: {
    width: '80%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 20,
    backgroundColor: COLORS.primary,
    borderRadius: 5
  },
  buttonText: {
    fontSize: SIZE.buttonSize,
    fontFamily: FONTS.primary,
    color: 'white'
  }
});

export default styles;
