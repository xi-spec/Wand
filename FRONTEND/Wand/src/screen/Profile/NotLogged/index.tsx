import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

export default function NotLogged ({ navigation }:{navigation:{navigate:Function}}) {
  return (
      <View style={styles.container}>

        <Text style={styles.text}>
            I alredy have a user account
        </Text>

        <TouchableOpacity
        testID='login'
        style ={styles.buttonContainer}
         onPress={() => navigation.navigate('Login')}>
            <Text style ={styles.buttonText}>LOG IN</Text>
        </TouchableOpacity>
<View style={styles.line}></View>
        <Text style={styles.text}>
           New user
        </Text>

        <TouchableOpacity
         testID='register'
        style ={styles.buttonContainer}
         onPress={() => navigation.navigate('Register')}>
            <Text style ={styles.buttonText}>REGISTER</Text>
        </TouchableOpacity>

      </View>
  );
}
