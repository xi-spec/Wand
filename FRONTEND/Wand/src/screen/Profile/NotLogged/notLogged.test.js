
import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import NotLogged from './index';

describe('Given a NotLogged component', () => {
  let navigation;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn()
    };
  });
  describe('When is rendered', () => {
    test('Then is matched to snapshop', () => {
      const tree = render(<NotLogged navigation={navigation} />);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When LOGIN button is pressed" ', () => {
    test('Then navigation navigate is invoked"', () => {
      const { getByTestId } = render(<NotLogged navigation={navigation}/>);

      fireEvent.press(getByTestId('login'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When REGISTER button is pressed" ', () => {
    test('Then navigation navigate is invoked"', () => {
      const { getByTestId } = render(<NotLogged navigation={navigation}/>);

      fireEvent.press(getByTestId('register'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
