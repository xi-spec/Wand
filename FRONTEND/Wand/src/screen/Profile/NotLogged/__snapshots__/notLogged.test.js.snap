// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NotLogged component When is rendered Then is matched to snapshop 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 18,
        "marginVertical": 20,
      }
    }
  >
    I alredy have a user account
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2E3192",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="login"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "MontserratRegular",
          "fontSize": 20,
        }
      }
    >
      LOG IN
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#E6E6E6",
        "borderBottomWidth": 1,
        "marginLeft": 5,
        "marginRight": 5,
        "width": "80%",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 18,
        "marginVertical": 20,
      }
    }
  >
    New user
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2E3192",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="register"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "MontserratRegular",
          "fontSize": 20,
        }
      }
    >
      REGISTER
    </Text>
  </View>
</View>
`;
